src/pint/models/model_builder.py

```diff
@@ -151,6 +164,8 @@ def __call__(
         convert_tcb = allow_tcb == True
         allow_tcb_ = allow_tcb in [True, "raw"]
 
+        assert allow_T2 in [True, False]
```

Comment:
The docstring says this can also be an str

Reply:
Isn't it better to say `isinstance(allow_T2, bool)`?

Reply:
Yes, changing